src/javascript/app/pages/trade/digit_ticker.js

```diff
@@ -0,0 +1,124 @@
+const DigitTicker = (() => {
+    let $container,
+        $peek,
+        $peek_box,
+        $mask,
+        total_tick_count,
+        contract_status,
+        current_spot;
+
+    const digit_block_size        = 36;
+    const array_of_digits         = [0, 1, 2, 3, 4, 5, 6, 7, 8, 9];
+    const style_offset_correction = 2;
+
+    // Calculate left margin of the peek-box against the "width" of the container
+    const calculateLeftMargin = (width) => (width - (digit_block_size * 10)) / 2;
+
+    // Calculate peek-box left offset.
+    const calculateOffset = () => {
+        const left_offset = calculateLeftMargin($container.offsetWidth);
+        return (+current_spot * digit_block_size) + left_offset - style_offset_correction;
+    };
+
+    const fail = () => {
```

Comment:
I think this should be moved closer to the success function.

Reply:
all methods are ordered alphabetically. Should I not do that?

Reply:
not really. generally we order them with init being the first function, any onunload sort of function at the bottom, and then the most relevant functions together in between.